src/mdlComponentHandler.js

```diff
@@ -86,6 +86,15 @@ var componentHandler = (function() {
    * the element to.
    */
   function upgradeElementInternal(element, jsClass) {
+    if (jsClass === undefined) {
+      for (var i = 0; i < registeredComponents_.length; i++) {
```

Comment:
re: IE9 @addyosmani can probably comment further on this since I wasn't here when this happened. But definitely now JS is disabled for IE<10. See: https://github.com/google/material-design-lite/blob/master/src/mdlComponentHandler.js#L278

Basically a we check that a few functions are available and since we check for DomElement.classList this rules out IE9...